common/websocket/index.ts

```diff
@@ -41,52 +42,79 @@ class WebSocketService {
         return this.connectedClients;
     }
 
-    private addClient(id: string, connection: WebSocket) {
-        this.connectedClients[id] = connection;
+    private addClient(userId: string, connection: WebSocket) {
+        (this.connectedClients[userId] = this.connectedClients[userId] || []).push(connection);
     }
 
     private removeClient(id: string) {
+        Object.keys(this.connectedClients).forEach((key) => {
+            const idx = this.connectedClients[key].findIndex((ws: WebSocket & { clientId: string }) => ws.clientId === id);
+            if (idx !== -1) {
+                this.connectedClients[key].splice(idx, 1);
+            }
+            if (this.connectedClients[key].length === 0) {
+                delete this.connectedClients[key];
+            }
+        });
         delete this.connectedClients[id];
```

Comment:
not sure about this line - shouldn't the array be only indexed by userId?

Reply:
I just forgot to delete this line. Thank's for your attention.
I will remove that line.